.github/workflows/iroha2-dev-pr-static.yml

```diff
@@ -17,25 +17,28 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  check:
+  analysis:
     runs-on: ubuntu-latest
     container:
       image: 7272721/i2-ci:nightly
     steps:
       - uses: actions/checkout@v3
       - uses: Swatinem/rust-cache@v2
 
-      - name: Format check
+      - name: Format
         run: cargo +nightly-2022-08-15 fmt --all -- --check
       - name: Lints without features
-        run: cargo +nightly-2022-08-15 lints clippy --workspace --benches --tests --examples --quiet --no-default-features
         if: always()
+        run: cargo +nightly-2022-08-15 lints clippy --workspace --benches --tests --examples --quiet --no-default-features
       - name: Lints with all features enabled
+        if: always()
         run: cargo +nightly-2022-08-15 lints clippy --workspace --benches --tests --examples --quiet --all-features
+      - name: Documentation without features
         if: always()
-      - name: no_std compatibility check
-        run: cargo nono check --package iroha_data_model --no-default-features
+        run: cargo +nightly-2022-08-15 doc --no-deps --quiet --no-default-features
+      - name: Documentation with all features enabled
         if: always()
-      - name: Documentation check
-        run: if cargo doc -q 2>&1 | grep '^'; then exit 1; fi
+        run: cargo +nightly-2022-08-15 doc --no-deps --quiet --all-features
+      - name: no_std compatibility
```

Comment:
These shouldn't be necessary. We are using conditional compilation where we should instead be using LTO.

Reply:
Are you suggesting to remove `Documentation with all features enabled` and `no_std compatibility` checks?

Reply:
Just the `--all-features` documentation.

Reply:
Can be done in a separate PR.

Reply:
#2663

Reply:
Done in this PR